server/controllers/users.js

```diff
@@ -0,0 +1,296 @@
+import * as _ from 'lodash';
+import randomstring from 'randomstring';
+import { User, Message } from '../models';
+import { transporter, helperOptions } from '../config/nodemailer';
+
+// Function to signup new users
+export const signup = (req, res) => {
+  const username = req.body.username.trim().toLowerCase();
+  const email = req.body.email.trim().toLowerCase();
+  User.create({
+    username,
+    password: req.body.password,
+    email,
+  })
+  .then((user) => {
+    const token = user.generateAuthToken();
+    res.header('x-auth', token).status(201).send({
+      message: `welcome ${user.username}`,
+      user
+    });
+  })
+  .catch(() => res.status(400));
+};
+
+// Function to sign users in
+export const signin = (req, res) => {
+  const body = _.pick(req.body, ['username', 'password']);
+  const username = body.username.trim().toLowerCase();
+  if (!body.password) {
+    return res.status(400).json({
+      error: 'Password must not be empty'
```

Comment:
I will suggest you try as much as possible to be consistent with your design pattern, I can see you are using a middleware to validate the username, but you seem to be validating the password inside controller.